src/com/goide/inspections/GoStringIndexIsByteInspection.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.GoTypes;
+import com.goide.psi.*;
+import com.goide.psi.impl.GoTypeUtil;
+import com.goide.quickfix.GoStringIndexIsByteQuickFix;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.util.Trinity;
+import com.intellij.psi.tree.TokenSet;
+import org.jetbrains.annotations.NotNull;
+
+public class GoStringIndexIsByteInspection extends GoInspectionBase {
+
+  private static final String TEXT_HINT = "Mismatched types: byte and string";
+  private static final GoStringIndexIsByteQuickFix STRING_INDEX_IS_BYTE_QUICK_FIX = new GoStringIndexIsByteQuickFix();
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+
+      @Override
+      public void visitConditionalExpr(@NotNull GoConditionalExpr o) {
+        GoExpression left = o.getLeft();
+        GoExpression right = o.getRight();
+
+        GoIndexOrSliceExpr indexExpr;
+        GoStringLiteral stringLiteral;
+        if (left instanceof GoIndexOrSliceExpr && right instanceof GoStringLiteral) {
```

Comment:
It doesn't seem as clean code. At least reuse ```getFirstStringLiteral```.
also good idea to introduce method like ```getFirstElementOfType(Class clazz, PsiElement ...elements)``` in ```GoPsiImplUtil```